account_invoice_force_number/views/invoice_view.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data>
+
+    <record model="ir.ui.view" id="account_invoice_form_int_number">
+        <field name="name">account.invoice.form.int_number</field>
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_form"/>
+        <field name="arch" type="xml">
+            <field name="journal_id" position="after">
+                <field name="internal_number"
+                    attrs="{'invisible': [('state', '!=', 'draft')]}"/>
+            </field>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="account_invoice_supplier_form_int_number">
+        <field name="name">account.invoice.supplier.form.int_number</field>
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_supplier_form"/>
+        <field name="arch" type="xml">
+            <field name="journal_id" position="after">
+                <field name="internal_number"
+                    attrs="{'invisible': [('state', '!=', 'draft')]}"/>
```

Comment:
I'm looking at the diff with version 8: http://pastie.org/10819678
Please keep old attributes, like ```help``` and ```groups```